packages/ckeditor5-link/tests/linkcommand.js

```diff
@@ -628,6 +633,62 @@ describe( 'LinkCommand', () => {
 
 					expect( command.value ).to.equal( 'url' );
 				} );
+
+				describe( 'keep formatting attributes', () => {
```

Comment:
I'd add some tests with the selection collapsed in a link and different formatting attributes mixed to make sure the link range is found properly and updated. Especially with the same URL but changed text.